### DISCUSSION TITLE
How to horizontally scale Milvus clusters using Kubernetes to expand Bookies？

### DISCUSSION THREAD

**Discussion by Gy1900** [milvus-io/milvus#34986] [Q&A and General discussion]
Requirements:
I previously experienced a crash in my Milvus cluster with an error indicating that the Bookie memory was insufficient, so I want to scale out the Bookies. The error message was as follows: “Not enough bookies to create ledger with ensembleSize=2, writeQuorumSize=2 and ackQuorumSize=2.”

Background:
![collection](https://github.com/user-attachments/assets/539d054e-fcef-443e-888c-c0eef61fb974)
I am using a 6-server setup with three primary and three secondary machines (each with 8C16G) and Kubernetes to construct 26 nodes for deploying the Milvus cluster.

Objective:
I now need to scale out horizontally to support more data. How do I go about this? In this document https://milvus.io/docs/scaleout.md, I only saw the Helm method, but I did not see the Kubernetes operations.

Also, I have a question about my storage structure. The actual number of entities that can be stored is much less than the anticipated amount, and the service crashes. Could it be because I am also storing text? Do you have any suggestions in this regard?
![attu](https://github.com/user-attachments/assets/d7c53ca1-d1f3-4b40-8d66-85be691c1178)
![collection](https://github.com/user-attachments/assets/ef3ef33c-a9be-4694-a0a3-c63fb1dbe784)
![collection-col](https://github.com/user-attachments/assets/d25df4c5-9225-4178-a4ce-82d1db5b4f71)
![pods](https://github.com/user-attachments/assets/b03adb33-286f-4e36-9c65-a259367535b9)

**Comment by xiaofan-luan** [comment]
are you using Operator?

https://github.com/zilliztech/milvus-operator/blob/main/docs/administration/manage-dependencies/message-storage.md

please check if that help you.

You can change bookkeeper replica count to 3 and add more memory and cpu to it
